### ISSUE TITLE
About reaction generation task

### ISSUE THREAD

**Issue by BruceMa-1999** [gohar-malik/InterMask#9]
Hi~

That is a great work, and I have learned so much from it!

I notice you upload the best model of human-interaction, could you please upload the model of reaction-generation work?

I tried to train it on my own device, but I only have 3090 so that I cannnot run it.

Thank you!

**Comment by gohar-malik** [comment]
Hi Bruce,

Thank you for your interest in InterMask and I am glad to hear that you found it insightful.

InterMask does not require any separate training for the reaction generation task. In fact, this flexibility is one of the key novelties of InterMask. The same model trained on the interaction generation task can be used to generate reactions without any retraining or fine-tuning.

The only change happens at inference time:
- We extract the first person’s tokens from the VQ-VAE encoder.
- During generation, the first person’s tokens remain fully unmasked and are never regenerated.
- The second person’s tokens are initialized as fully masked, and the model progressively fills them in conditioned on the fixed first-person sequence.

All the reaction-generation results shown in the paper and on the project website are obtained using this inference-time modification only, with the model trained on the standard interaction generation task.

If you'd like to reproduce some reaction-generation results, you can run:
`python eval.py --gpu_id 0 --dataset_name interhuman --name trans_default --gen_react --time_steps 12 --mm_num_samples 0 --save_vis True`

**Comment by BruceMa-1999** [comment]
> Hi Bruce,
> 
> Thank you for your interest in InterMask and I am glad to hear that you found it insightful.
> 
> InterMask does not require any separate training for the reaction generation task. In fact, this flexibility is one of the key novelties of InterMask. The same model trained on the interaction generation task can be used to generate reactions without any retraining or fine-tuning.
> 
> The only change happens at inference time:
> 
> * We extract the first person’s tokens from the VQ-VAE encoder.
> * During generation, the first person’s tokens remain fully unmasked and are never regenerated.
> * The second person’s tokens are initialized as fully masked, and the model progressively fills them in conditioned on the fixed first-person sequence.
> 
> All the reaction-generation results shown in the paper and on the project website are obtained using this inference-time modification only, with the model trained on the standard interaction generation task.
> 
> If you'd like to reproduce some reaction-generation results, you can run: `python eval.py --gpu_id 0 --dataset_name interhuman --name trans_default --gen_react --time_steps 12 --mm_num_samples 0 --save_vis True`

Hi Gohar,

It’s a great honor to receive your reply — it has been tremendously helpful to me.

I evaluated the model following your duidance, but I got the FID of 19.6046 instead of the 2.99 in the paper.

Is there anything I might have done incorrectly?

========== MM Distance Summary ==========
---> [ground truth] Mean: 3.7841 CInterval: 0.0008
---> [react_best_fid.tar] Mean: 3.9430 CInterval: 0.0015
========== R_precision Summary ==========
---> [ground truth](top 1) Mean: 0.4259 CInt: 0.0048;(top 2) Mean: 0.6044 CInt: 0.0040;(top 3) Mean: 0.7078 CInt: 0.0036;
---> [react_best_fid.tar](top 1) Mean: 0.1595 CInt: 0.0042;(top 2) Mean: 0.2527 CInt: 0.0049;(top 3) Mean: 0.3182 CInt: 0.0052;
========== FID Summary ==========
---> [ground truth] Mean: 0.2919 CInterval: 0.0058
---> [react_best_fid.tar] Mean: 19.6046 CInterval: 0.2176
========== Diversity Summary ==========
---> [ground truth] Mean: 7.7718 CInterval: 0.0281
---> [react_best_fid.tar] Mean: 7.2724 CInterval: 0.0278
========== MultiModality Summary ==========

**Comment by gohar-ch** [comment]
Hi Bruce,

I tested the evaluation script on my end got the following results using the command `python eval.py --gpu_id 0 --dataset_name interhuman --name trans_default --gen_react --time_steps 12 --mm_num_samples 0 `

```
========== MM Distance Summary ==========
---> [ground truth] Mean: 3.7847 CInterval: 0.0008
---> [react_best_fid.tar] Mean: 3.7801 CInterval: 0.0011
========== R_precision Summary ==========
---> [ground truth](top 1) Mean: 0.4204 CInt: 0.0051;(top 2) Mean: 0.6009 CInt: 0.0058;(top 3) Mean: 0.7041 CInt: 0.0053;
---> [react_best_fid.tar](top 1) Mean: 0.4559 CInt: 0.0053;(top 2) Mean: 0.6181 CInt: 0.0059;(top 3) Mean: 0.7051 CInt: 0.0052;
========== FID Summary ==========
---> [ground truth] Mean: 0.2863 CInterval: 0.0070
---> [react_best_fid.tar] Mean: 3.0214 CInterval: 0.0400
========== Diversity Summary ==========
---> [ground truth] Mean: 7.7762 CInterval: 0.0195
---> [react_best_fid.tar] Mean: 7.7209 CInterval: 0.0261
========== MultiModality Summary ==========
```
I am not sure what might be the issue on your end. Could you try cloning a new clean version of the code and running the evaluation again?

I have attached the full evaluation result file for your reference here:
[react_evaluation_best_fid_ts12_cs2_topkr0.9.log](https://github.com/user-attachments/files/24267654/react_evaluation_best_fid_ts12_cs2_topkr0.9.log)
